src/KernelDensity.jl

```diff
@@ -5,15 +5,39 @@ using StatsBase
 using Distributions
 using Optim
 using Interpolations
+using Flux.Tracker: TrackedReal
+using Flux
 
 import StatsBase: RealVector, RealMatrix
 import Distributions: twoπ, pdf
 import FFTW: rfft, irfft
+import Flux.Tracker: conv
+import Base.round
+import Core.Integer
 
 export kde, kde_lscv, UnivariateKDE, BivariateKDE, InterpKDE, pdf
 
 abstract type AbstractKDE end
 
+"""n-dimensional convolution"""
+function conv(x::AbstractArray{T,N}, w::AbstractArray{T,N}) where {T,N}
+	wdim = Int.(ceil.((size(w).-1)./2))
+	padding = Iterators.flatten([ (wdim[i],wdim[i]) for i=1:length(wdim) ]) |> collect
+
+	dims = DenseConvDims((size(x)...,1,1),(size(w)...,1,1); padding=padding )
+	result = Tracker.conv( reshape(x,(size(x)...,1,1)), reshape(w,(size(w)...,1,1)), dims)
+	return dropdims(result, dims = (1+N,2+N))
+end
+
+# patches for TrackedReal and Vector{TrackedReal}
```

Comment:
all of these methods below are [type piracy](https://docs.julialang.org/en/v1/manual/style-guide/index.html#Avoid-type-piracy-1), so probably better to make a PR to [Tracker.jl](https://github.com/FluxML/Tracker.jl)

Reply:
I see.. how to rewrite the `conv` methods in `univariate.jl` and  `bivariate.jl`?